src/code/z_kankyo.c

```diff
@@ -922,54 +962,68 @@ void Environment_Update(GlobalContext* globalCtx, EnvironmentContext* envCtx, Li
             CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_kankyo.c", 1690);
         }
 
-        if ((envCtx->unk_BF != 0xFF) && (envCtx->unk_DC != 2) && (envCtx->unk_BD != envCtx->unk_BF) &&
-            (envCtx->unk_D8 >= 1.0f) && (envCtx->unk_BF < 0x20)) {
-            envCtx->unk_BE = envCtx->unk_BD;
-            envCtx->unk_BD = envCtx->unk_BF;
-            envCtx->unk_D8 = 0.0f;
+        if ((envCtx->lightSettingOverride != LIGHT_SETTING_OVERRIDE_NONE) &&
+            (envCtx->lightBlendOverride != LIGHT_BLEND_OVERRIDE_FULL_CONTROL) &&
+            (envCtx->lightSetting != envCtx->lightSettingOverride) && (envCtx->lightBlend >= 1.0f) &&
+            (envCtx->lightSettingOverride <= LIGHT_SETTING_MAX)) {
+
+            envCtx->prevLightSetting = envCtx->lightSetting;
+            envCtx->lightSetting = envCtx->lightSettingOverride;
+            envCtx->lightBlend = 0.0f;
         }
 
-        if (envCtx->unk_BF != 0xFE) {
-            if (!envCtx->indoors && (envCtx->unk_BF == 0xFF)) {
-                for (i = 0; i < ARRAY_COUNT(D_8011FB48[envCtx->unk_1F]); i++) {
-                    if ((gSaveContext.skyboxTime >= TIME_ENTRY_1F.startTime) &&
-                        ((gSaveContext.skyboxTime < TIME_ENTRY_1F.endTime) || TIME_ENTRY_1F.endTime == 0xFFFF)) {
+        if (envCtx->lightSettingOverride != LIGHT_SETTING_OVERRIDE_NONE - 1) {
```

Comment:
the -1 is because cutscene commands subtract 1 from the index
`globalCtx->envCtx.lightSettingOverride = cmd->setting - 1;`

I have to refresh myself on the override stuff in general to have more thoughtful opinions, but this is where the -1 comes from at least 

presumably a cutscene would pass 0xFF when it wants to use the light setting from the floor poly, and the cutscene system will subtract 1 to make that 0xFE